modules/tutorials/pages/java-tutorial/retrieve-records.adoc

```diff
@@ -0,0 +1,275 @@
+
+==== Retrieve All Course Records
+
+To retrieve all of your course records using the query editor:
+
+. Enter the following query into your query editor:
++
+[source, sqlpp]
+----
+select crc.* from `course-record-collection` crc
+----
++
+. Click btn:[Execute] to retrieve all course records.
++
+[source, json]
+----
+[
+  {
+    "course-name": "art history",
+    "credit-points": 100,
+    "faculty": "fine art"
+  },
+  {
+    "course-name": "fine art",
+    "credit-points": 50,
+    "faculty": "fine art"
+  },
+  {
+    "course-name": "graphic design",
+    "credit-points": 200,
+    "faculty": "media and communication"
+  }
+]
+----
+
+==== Retrieve Course Records with Less than 200 Credits
+
+You can expand your query to narrow your search down further.
+To retrieve only courses with less than 200 `credit-points` using the query editor:
+
+. Enter the following query into your query editor:
++
+[source, sqlpp]
+----
+select crc.* from `course-record-collection` crc where crc.`credit-points` < 200
+----
++
+. Click btn:[Execute] to retrieve all courses with less than 200 credits.
++
+[source, json]
+----
+[
+  {
+    "course-name": "art history",
+    "credit-points": 100,
+    "faculty": "fine art"
+  },
+  {
+    "course-name": "fine art",
+    "credit-points": 50,
+    "faculty": "fine art"
+  }
+]
+----
+
+=== Retrieve Record IDs
+
+The `id` field is not automatically returned when you retrieve all of your course information.
+
+The `id` is part of a document's meta structure, and to retrieve it you must adjust your {sqlpp} query and run it again:
+
+. Enter the following query into your query editor:
++
+[source, sqlpp]
+----
+select META().id, crc.* from `course-record-collection` crc where crc.`credit-points` < 200
+----
++
+The `META()` function call returns any property contained inside the document's metadata, including the ID.
++
+. Click btn:[Execute] to retrieve course records and their IDs.
++
+[source, json]
+----
+[
+  {
+    "course-name": "art history",
+    "credit-points": 100,
+    "faculty": "fine art",
+    "id": "ART-HISTORY-000001"
+  },
+  {
+    "course-name": "fine art",
+    "credit-points": 50,
+    "faculty": "fine art",
+    "id": "FINE-ART-000002"
+  }
+]
+----
+
+
+[#retrieve-with-sdk]
+== With the SDK
+
+You can also use {sqlpp} queries to retrieve your records with the SDK.
+Unlike the query editor, though, you must include the name of the bucket and the scope to fully qualify the name of the collection in the {sqlpp} statement in your application. For example:
+
+[source, sqlpp]
+----
+select crc.* from `student-bucket`.`art-school-scope`.`course-record-collection` crc
+----
+
+=== Retrieve Your Records
+
+==== Retrieve All Course Records
+
+To retrieve all of your course records using the Java SDK:
+
+. In your `java` directory, create a new file called `ArtSchoolRetriever.java`.
+. Paste the following code block into your `ArtSchoolRetriever.java` file:
++
+[source, java]
+----
```

Comment:
Same thing about sample code. For longer samples, it's better to pull in via include - for the single line examples, less important (but still something to consider)